internal/backoff/backoff.go

```diff
@@ -30,49 +30,156 @@ import (
 
 // Strategy defines the methodology for backing off after a grpc connection
 // failure.
-//
 type Strategy interface {
 	// Backoff returns the amount of time to wait before the next retry given
 	// the number of consecutive failures.
 	Backoff(retries int) time.Duration
+	// MinConnectionTimeout returns the minimum amount of time we are willing to
+	// give a connection to complete.
+	MinConnectionTimeout() time.Duration
```

Comment:
MinConnectTimeout is not a function of backoff, so it doesn't really belong in this package.  I think we can keep MCT separate from backoff strategy and store them separately in the ClientConn.

Reply:
Removed it from here.
Added it as a field in `dialOptions` with `defaultDialOptions()` setting the existing default value for this. Overrides are done from `WithConnectParams()`, which feels a little odd to me, but thought I will run it by you anyways.